### DISCUSSION TITLE
Rdio-Scanner & SDRTrunk Setup Issues

### DISCUSSION THREAD

**Discussion by ramarco** [chuot/rdio-scanner#418] [Q&A] [answered]
I posted this question on RadioReference but haven't gotten any replies. Can anyone here help?

I've enjoyed SDRTrunk for a few months and would like to listen to it on my phone and work computer. Rdio-Scanner seems like a good way to do this but the setup has me feeling defeated. I had it partially working but only 2 of my talkgroups were playing on Rdio-Scanner. I spent hours trying to figure out why but finally decided to delete everything and start with a fresh copy of Rdio-Scanner. After setting it up again, nothing plays on Rdio-Scanner. So far the only things I can find that seem wrong are the following:

1) In Rdio-Scanner - Config - ApiKey, I click on the system I'm trying to set up - Michigan's Public Safety Communications System (MPSCS) - and then click on "Choose Systems". At the top, the "Everything" box is checked as well as all the default groups and tags. At the bottom, under "Systems", there is a box labeled "Michigan's Public Safety Communications System (MPSCS)". I check this box, click "Ok" and then click "Save". When I go back to confirm it is saved, the box is no longer checked. So, it appears that it isn't saving.

2) The setting for "Disable the API key" always says "Disabled" no matter which way the toggle is set.

3) The following message keeps coming up on the server window:  api: Incomplete call data: no talkgroup

I've been working on this on and off for a week now. When I have downtime at work, I read and reread all the posts I can find on this subject. I read that many people struggle to set up Rdio-Scanner but love it once they get it to work. I found several written and YouTube guides for setting up SDRTrunk but sadly they don't seem to exist for Rdio-Scanner. I'd be grateful for any help getting this to work.

**Comment by ramarco** [comment, marked as answer]
Another day spending hours on this, much to my wife's displeasure, but I believe the problem(s) is solved. I'll relay my findings in case anyone else struggles with this.

1) The setting for "Disable the API key" needs to be toggled OFF. I thought ON would mean the API key is active and OFF would mean that it is not. Apparently, ON means that the API key is OFF. And it still says "Disabled" when the API key is ON. It seems counterintuitive to me, but I'm not a programmer.

2) I thought that toggling ON the Auto Populate setting would create the talkgroups for me since it says, "Allows the automatic creation of unconfigured talkgroups." However, I had to download a CSV file of my local talkgroups from RadioReference and import them into Rdio-Scanner. This seems to have corrected the "api: Incomplete call data: no talkgroup" error.

I'm now able to listen to SDRTrunk while on the go using Rdio-Scanner. This is a great program that I will support if it continues to work properly. However, it needs better documentation. Perhaps if you're smart enough to create a program like this, you assume everyone is smart enough to figure it out on their own.

**Comment by srs4511351** [comment]
I saw your post and it sounds like you had some of the same problems that I am having. I am also using Michigan's Public Safety Communications System (MPSCS).

I am using a Raspberry Pi computer with Trunk Recorder, which I believe is working correctly, as it is making wav files recordings of the talkgroups. I have imported the 6000 or so MPSCS talkgroups to it.

I also have SDRtrunk, but being written in java, it hogs way too much CPU power to work well.
I have some questions about your post.

1. When you toggled "Disable the API key" to OFF, which way with the graphic "switch"? Was the "dot" to the right or the left? I tried several combinations and managed to get MPSCS to stay checked in "Choose Systems", but the Everything box was checked and the Groups boxes were unchecked. I did see the problem you sere having.
2. Where were you seeing the error: "api: Incomplete call data: no talkgroup"? SDTtrunk or Rdio Scanner? Rdio Scanner recommends against importing more than 800 talkgroups and MPSCS has more that 6000.

I don't think that Trunk Recorder and Rdio Scanner are talking. Do you know how the communication is specified? I don't see it in setup admin.

In the Trunk Recorder terminal window, I see:
```
[2024-03-06 23:46:08.177666] (error)   [MPSCS]	819C	TG:      12303	Freq: 855.812500 MHz	Rdio Scanner Upload Error: 
[2024-03-06 23:46:08.177877] (error)   Plugin Manager: call_end -  rdioscanner_uploader failed
[2024-03-06 23:46:09.007601] (error)   [MPSCS]	819C 	TG:      12303	Wed Mar  6 23:45:54 2024 EST retry attempt 1 in 63s	 retry queue: 13 calls
[2024-03-06 23:46:14.015166] (error)   [MPSCS]	796C	TG:       2148	Freq: 855.187500 MHz	Rdio Scanner Upload Error: 
[2024-03-06 23:46:14.017303] (error)   Plugin Manager: call_end -  rdioscanner_uploader failed
[2024-03-06 23:46:15.007610] (error)   [MPSCS]	796C Failed to conclude call - TG:       2148	Wed Mar  6 23:42:53 2024 EST
```
There are no messages in the Rdio Scanner terminal window except those related to the web browser client.

**Reply by srs4511351** [reply]
I figured out the problem with communication.
I had to specify port 3000 in the Trunk Recorder config.json file. Starting Rdio Scanner like this also worked.
`sudo ./rdio-scanner -listen :80`

I did not have to import the talkgroups into Rdio Scanner.
"Disable the API key." switch is slid to the left, which is off, meaning do not disable the API key.

Under Systems, Auto Populate is slid to the right, turning it on.
The talkgroups are filling in.

It seems to be working well but I am having problems covering the frequency range. It seems that Trunk Recorder need to have the SDR bandwidth wide enough to cover the 11 MHz range that covers all of the channels.
op25 is smart enough to move the center frequency as needed.

The low and high frequencies are not covered with a 10 MHz SDR bandwidth:
```
[2024-03-07 01:09:24.131246] (error)   [MPSCS]	277C	TG:      22652	Freq: 861.262500 MHz	Not Recording: no source covering Freq
[2024-03-07 01:09:29.625563] (error)   [MPSCS]	278C	TG:        188	Freq: 851.175000 MHz	Not Recording: no source covering Freq
```

**Comment by ramarco** [comment]
@srs4511351 - It looks like you answered your own questions, which is what I had to do. There doesn't seem to be much active support for Rdio-Scanner, but it is free and developers get busy with real life.

The "Disable the API key" toggle was very confusing to me but I did finally figure it out through trial and error.

You said SDRTrunk hogs way too much CPU power to work well. That may be because you imported all 6000 or so MPSCS talkgroups to it. I have one playlist with everything and one with just my local stuff, which is what I usually use. I think it runs a bit slower when I use the "Everything" playlist. That's on a Dell Optiplex with an i7 CPU and 16 GB of RAM. It sounds like op25 is working better for you so I guess it doesn't matter.

I believe my needed bandwidth is covered with the device I'm using. It's basically two dongles in an aluminum case for passive thermal heat dissipation. I haven't seen any error messages saying otherwise.

**Comment by srs4511351** [comment]
Yes, I have done a lot of figuring things out. Thanks for helping.
That is what my technology web site is all about. I usually start with knowledge from the developer's web site, the rest of the web, then figuring it out myself.

SDRtrunk is slow and consumes too much CPU power because it is written in Java. Java is not an efficient language.
I just made a new installation of SDRtrunk and I did not import any talkgroups. It is working as I type, but the CPU is at 95%. I usually disable the spectrum and waterfall display. That helps a lot.
By stark contrast, op25 has all of the talkgroups and only uses 25%-30% CPU! You should give it a try unless you are contributing to services like broadcastify. I give the complete MPSCS setup for Washtenaw county. Also, the University on Michigan.

I have 3 different SDRs, but I only need the one HackRF. However, rdio-scanner does not allow sample rates greater than 10 MHz. I should open an issue.

**Comment by Thinline-Dynamic-Solutions** [comment]
I run a 24 core cpu no issues


*Michael Chambers - Ohio Radio Scanner Network*
Founder/Manager/Support
***@***.**** ***@***.***>

**Comment by srs4511351** [comment]
I can make a good scanner on a Raspberry Pi with only 4 cores using op25 or trunk-recorder for $200-$250.

Inefficient code is a waste.
The HackRF only has an 8 bit ADC, so it runs at a lower CPU load.

**Reply by ramarco** [reply]
Thank you for the information. op25 sounds like a good fit for you. I'd try it but I'm a Broadcastify feed provider.

Fortunately, my CPU usage averages around 15% while running SDRTrunk, Rdio-Scanner, ScannerCast, and several other programs. Not sure why this is because I'm just using an older refurb computer with an i7 CPU and 16 GB of RAM that I bought from eBay for $150.

**Comment by crossterra-agent** [comment]
Requested completion review: https://github.com/CrossTerra/monorepo/pull/2257#issuecomment-4180020634
